azurelinuxagent/common/protocol/imds.py

```diff
@@ -263,7 +272,9 @@ def get_compute(self):
         resp = restutil.http_get(self.compute_url, headers=self._headers)
 
         if restutil.request_failed(resp):
-            raise HttpError("{0} - GET: {1}".format(resp.status, self.compute_url))
+            resp = restutil.http_get(self.compute_url_backup, headers=self._headers)
+            if restutil.request_failed(resp):
+                raise HttpError("{0} - GET: {1}".format(resp.status, self.compute_url))
```

Comment:
Wouldn't it be useful to include in the message that the fallback failed too? Or at least log it locally? Could help a lot for livesite.

Reply:
That's a great idea. I'll add that with the next iteration. Thanks!

Reply:
could we add the logging on this iteration? thanks

Reply:
I meant on my next push to this PR. I need to fix the logic noted by my "TODO" comment before this is ready to merge.

Reply:
Thank you everyone for reviewing! I hope to have the next commit pushed sometime on Monday to address this plus the fallback logic.

Reply:
ah, ok! please ping me when all the changes are in and I can take a look to the PR. thanks

Reply:
Business logic is done and tested on an Azure VM. I'm figuring out how to mock the IOError. I'll ping you once I have the tests working.